### PR TITLE
N beats bugfix.

### PR DESCRIPTION
**Pull request by AaronSpieler**
*Issue #, if available:*
Fixes issue https://github.com/awslabs/gluon-ts/issues/629.

*Description of changes:*
Just named all the layers in the network explicitly.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mbohlkeschneider** [comment]
So just naming of layers, no actual bug?

File: src/gluonts/model/n_beats/_network.py [hunk 1]
```diff
@@ -447,6 +463,7 @@ def __init__(
                         if (block_id > 0 and sharing[stack_id])
                         else None
                     )
+                    # print(params)
```
Comment (mbohlkeschneider) [inline src/gluonts/model/n_beats/_network.py]: You forgot this here :-).

Comment (AaronSpieler) [reply src/gluonts/model/n_beats/_network.py]: yep :)

**Comment by AaronSpieler** [comment]
> So just naming of layers, no actual bug?

Well, it actually fixes a bug where if you specified for the generic model to share parameters, due to the automatic naming of the layers, mxnet wanted to use the parameters from the wrong layer. Naming them explicitly fixes this.

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/gluon-ts/pull/655?src=pr&el=h1) Report
> Merging [#655](https://codecov.io/gh/awslabs/gluon-ts/pull/655?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/gluon-ts/commit/8013024d826fe4117057d8fe82d98c23ce423f37&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.


[Impacted file tree graph](https://codecov.io/gh/awslabs/gluon-ts/pull/655?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #655   +/-   ##
=======================================
  Coverage   83.94%   83.94%           
=======================================
  Files         178      178           
  Lines       10337    10337           
=======================================
  Hits         8677     8677           
  Misses       1660     1660           
```

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/gluon-ts/pull/655?src=pr&el=h1) Report
> Merging [#655](https://codecov.io/gh/awslabs/gluon-ts/pull/655?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/gluon-ts/commit/8013024d826fe4117057d8fe82d98c23ce423f37&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.


[Impacted file tree graph](https://codecov.io/gh/awslabs/gluon-ts/pull/655?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #655   +/-   ##
=======================================
  Coverage   83.94%   83.94%           
=======================================
  Files         178      178           
  Lines       10337    10337           
=======================================
  Hits         8677     8677           
  Misses       1660     1660           
```

**Pull request merged**